CIPs/0007.md

```diff
@@ -0,0 +1,65 @@
+
+In order to promote a stable network, the Celo protocol includes slashing for downtime such that if a validator fails to participate in consensus for a given amount of time, they may be slashed. More information about slashing in the Celo protocol can be found on [docs.celo.org](https://docs.celo.org/celo-codebase/protocol/proof-of-stake/penalties)
+
+Currently, slashable downtime is defined as a window of blocks, greater than a predefined length, in which no block's parent aggregated seal includes a signature from the validator in question. It's currently implementation checks and executes slashing in a single transaction which requires an amount of gas linear in the required window size.
+
+Under its current implementation, the gas required to evaluate a downtime event may far exceed the block gas limit. In the Release Candidate 1 network, the downtime slashing window is configured to be 5760 blocks, which would require roughly 13.5 million gas to evaluate, while the block gas limit is set to 10 million.
+
+## Goal
+
+- Allow the user to slash a validator if it was down for a specific consecutive block window without running out of gas.
+
+## Proposed Solution
+
+Create a new contract funtion that divides the downtime verification in validation intervals.
+
+The user that wants to slash a particular validator will have to call a function from the contract that will generate a proof with the validators that signed at least one block for the specific interval set by the caller.
+
+The user will require to repeat this first process the amount of times necessary to cover with overlapping or adjacent intervals, at least the `slashableDowntime` amount of blocks set in the contract.
+
+Then the user must generate another contract call listing all the intervals required to verify that the validator to be slashed was down for those intervals, and that also covers the `slashableDowntime` window.
+
+### Note
+
+This process allows the user to validate a interval that will contain the data of every validator. This means that if for example, three validators where down in the same period, the same interval could be used to slash the three of them.
+The gas required to generate the `I_n` intervals will be spent only in the first validation.
+
+### Some numbers
+
+Tested with intervals of 5000 blocks (each one required less than 10 million gas), using 1000 intervals for the actual slash (required ~6.5 million gas) which covers 5M blocks.
+Even if is not necessary right now, this solution will cover possible governable incrementations of the `slashableDowntime` window.
+
+### Variation of the proposed solution
+
+- The slots validated could work for every user in the community. When a slashing is performed, every user that generated a validated slot that is used to demostrate the downtime, receives a portion of the reward. But to address this, changes are required in other contracts to calculate and distribute the rewards.
+
+## Alternative Solution
```

Comment:
I would remove this entire section. Alternative solutions should be proposed under their own respective CIPs

Reply:
I now see that Alternative Solutions was specified in the current template, so no worries, please keep this section :)